webCommander/centralHub/xTwitter/twitterAPI.py

```diff
@@ -0,0 +1,102 @@
+import os
+import requests 
+import json
+import config
+import pprint 
+from time import sleep   
+from requests_oauthlib import OAuth1Session                                                                                                                                                
+from dotenv import load_dotenv, find_dotenv
+from pathlib import Path
+
+load_dotenv()
+
+#DEFINE AUTHENTICATION TOKENS
+X_AUTHORIZATION_BEARER = os.getenv('X_BEARER_TOKEN')
+X_API_KEY = os.getenv('X_API_KEY')
+X_API_SECRET = os.getenv('X_API_SECRET')
+X_ACCESS_TOKEN = os.getenv('X_ACCESS_TOKEN')
+X_ACCESS_TOKEN_SECRET = os.getenv('X_ACCESS_TOKEN_SECRET')
+#X_AUTHENTICATION_URL = 'https://api.twitter.com/2/oauth2/token'
+
+#DEFINE URLS
+X_BASE_URL = 'https://api.twitter.com'
+X_GET_MYUSER = X_BASE_URL+'/2/users/me'
+X_GET_USER = X_BASE_URL+'/2/users/by/username/'
+X_TWEET_ID = X_BASE_URL+'/2/tweets/'
+X_GET_TWEET = X_BASE_URL+'/2/users/'
+
+#CODES
+TOO_MANY_REQUESTS = 429
+
+def x_authenticate():
+
+    return OAuth1Session(
+        X_API_KEY,
+        X_API_SECRET,
+        X_ACCESS_TOKEN,
+        X_ACCESS_TOKEN_SECRET
+    )
+
+def request_to_x_api(endpoint, headers={}, params={}):
+    x_session = x_authenticate()
+    response = x_session.get(endpoint, headers=headers, params=params)
+    while response.status_code == TOO_MANY_REQUESTS:
+        print('Too many requests, trying in 15 mins...')
+        sleep(900)
```

Comment:
decrease? when called inside a django view I think it would time out way before then, also I would define the int as a constant